qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -634,9 +634,21 @@ def on_mode_left(self, mode):
             widget = self.currentWidget()
             log.modes.debug("Left status-input mode, focusing {!r}".format(
                 widget))
-            if widget is None:
-                return
-            widget.setFocus()
+            if widget is not None:
+                widget.setFocus()
+
+        self._handle_clear_on_mode_leave(mode)
+
+    def _handle_clear_on_mode_leave(self, mode):
+        """Handle clearing page focus when leaving insert mode."""
+        if not config.val.input.blur_on_mode_leave or \
+           mode != usertypes.KeyMode.insert:
```

Comment:
Please use parenthesis for line continuation instead, and indent the second line more so it's clear that it's part of the condition:

```python
        if (not config.val.input.blur_on_mode_leave or
                mode != usertypes.KeyMode.insert):
```